test/spmd/test_train_spmd_linear_model.py

```diff
@@ -78,6 +77,14 @@ def train():
     train_loader = pl.MpDeviceLoader(
         train_loader, device, input_sharding=xs.ShardingSpec(mesh, (0, 1)))
 
+  if 'fsdp' in FLAGS.sharding:
+    train_loader = pl.MpDeviceLoader(
+        train_loader, device, input_sharding=xs.ShardingSpec(mesh, (0, 1)))
+    print('Sharding model weights')
+    # Shard the weights according to their 0th dim
+    xs.mark_sharding(model.fc1.weight, mesh, (0, 1))
+    xs.mark_sharding(model.fc2.weight, mesh, (0, 1))
```

Comment:
hmm we are using `mark_sharding` as an inplace op, if that's how we are going to let user uses it, we should make it `mark_sharding_` instead.

Reply:
Good catch. I feel like we probably shouldn't make it inplace op. Let me make a follow up to make it more clear.